eurocropsml/dataset/dataset.py

```diff
@@ -48,107 +49,164 @@ def __init__(
+                            try:
+                                cloud_probs = np.apply_along_axis(
+                                    partial(
+                                        find_clouds,
+                                        band4_idx=s2_data_bands.index("04"),
+                                        preprocess_config=preprocess_config,
+                                    ),
+                                    1,
+                                    data[idx : idx + count],
+                                )
+                            except ValueError:
+                                raise ValueError(
+                                    "Band 4 cannot be excluded if date_type is 'months'"
+                                )
+                            month_data[month] = data[idx + np.argmin(cloud_probs)]
+                    arrays_dict["data"][satellite] = month_data
+                    arrays_dict["dates"][satellite] = np.arange(12)
         return arrays_dict
 
     def __getitem__(
         self,
         idx: int,
     ) -> LabelledData:
-        f = self.file_list[idx]
+        f = {key: paths[idx] for key, paths in self.file_dict.items()}
+
         arrays_dict = self.mmap_store[f]
 
-        arrays_dict = self._format_dates(self.preprocess_config, self.config.date_type, arrays_dict)
+        arrays_dict = self._format_dates(
+            self.preprocess_config,
+            self.s2_data_bands,
+            self.config.date_type,
+            arrays_dict,
+        )
 
-        np_data = arrays_dict.pop("data")
+        np_data_dict = arrays_dict.pop("data")
         if self.keep_band_idxs is not None:
-            np_data = np_data[:, self.keep_band_idxs]
+            np_data_dict["S2"] = np_data_dict["S2"][:, self.keep_band_idxs]
 
-        meta_data: dict[str, torch.Tensor] = {
-            array_name: torch.tensor(np_array) for array_name, np_array in arrays_dict.items()
+        meta_data: dict[str, dict[str, torch.Tensor] | torch.Tensor] = {
+            array_name: {key: torch.tensor(np_array) for key, np_array in meta_array.items()}
+            for array_name, meta_array in arrays_dict.items()
         }
+        # center is always the same, replace by first value
+        meta_data["center"] = cast(torch.Tensor, next(iter(meta_data["center"].values())))
 
-        tensor_data = torch.tensor(np_data, dtype=torch.float)
+        # normalization and scaling to (0,1]
         if self.config.normalize:
-            tensor_data = torch.mul(tensor_data, NORMALIZING_FACTOR)
+            if "S1" in np_data_dict:
+                normalized_s1 = np_data_dict["S1"] * NORMALIZING_FACTOR_S1
+                np_data_dict["S1"] = (normalized_s1 + 1) / 2 + EPSILON
+            if "S2" in np_data_dict:
+                normalized_s2 = np_data_dict["S2"] * NORMALIZING_FACTOR_S2
+                np_data_dict["S2"] = normalized_s2 * (1 - EPSILON) + EPSILON
```

Comment:
@jmaces: I am thinking of removing the if-clause and just always apply the normalization. Otherwise, in case no normalization is applied, all our padding values etc. do not fit anymore.

Reply:
I agree that now it makes no sense to use our default padding without normalization. but I guess the padding could also be modified by, e.g., using a different custom `collate_fn`. Maybe instead of removing the option to deactivate normalization we could instead add a warning that this would require custom modification of the padding as well?